services/navigation/builtin/builtin.go

```diff
@@ -399,14 +400,17 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 			return svc.waypointReached(ctx)
 		}
 
-		// loop until no waypoints remaining
+		// do not exit loop - even if there are no waypoints
 		for {
+		WP:
 			if ctx.Err() != nil {
 				return
 			}
 
 			wp, err := svc.store.NextWaypoint(ctx)
-			if err != nil {
+			if err != nil && err.Error() == "no more waypoints" {
```

Comment:
i have to do `err != nil` first and have this conditional use the logical and operator since we want to avoid "nil pointer dereference"